### PR TITLE
Added terminal executor for vim/nvim

### PR DESCRIPTION
**Pull request by ilyachur**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -105,7 +105,12 @@ The options below allow to change plugin behavior.
  - **`g:cmake_compile_commands`** if this variable is not equal 0, plugin will generate compile commands data base. Default is 0.
  - **`g:cmake_compile_commands_link`** set the path for a link on compile_commands.json. Default is empty.
  - **`g:cmake_vimspector_support`** enables generation and modification of [Vimspector](https://github.com/puremourning/vimspector) config file. Default is 0. **Attention! The support of Vimspector config is an experimental feature.**
- - **`g:cmake_build_executor`** allows to force set the build executor. Available values are 'job', 'dispatch', 'system' and ''. Default is empty.
+ - **`g:cmake_build_executor`** allows to force set the build executor. Default is empty. Available values are:
+    - 'job' uses job to asynchronous build
+    - 'term' uses terminal to asynchronous build
+    - 'dispatch' uses [vim-dispatch](https://github.com/tpope/vim-dispatch) plugin to asynchronous build
+    - 'system' uses synchronous build
+    - '' uses automatic detection of supported modes
```
Comment (mark2185) [inline README.md:113]: In which order does it detect the modes? `job -> term -> dispatch`?

File: autoload/utils/exec/term.vim [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    "     echon 'Success! ' . l:cmd
+    " else
+    "     echon 'Failure! ' . l:cmd
+    " endif
+endfunction
+" }}} nvim functions "
+" }}} Private functions "
+
+function! utils#exec#term#run(cmd, open_qf, err_fmt) abort
+    let l:cmake4vim_term = 'cmake4vim_execute'
```
Comment (mark2185) [inline autoload/utils/exec/term.vim:86]: This can be declared after the `if` or not at all since it's used only once.

Comment (ilyachur) [reply autoload/utils/exec/term.vim]: Yes, I refactor this code.

File: README.md (hunk 1 above)
Comment (ilyachur) [reply README.md:113]: `dispatch`->`job`->`term`

Comment (ilyachur) [reply README.md]: Added the order

**Pull request merged**